brion/plugin/spikeReportBinary.cpp

```diff
@@ -55,85 +53,68 @@ class Header
     uint32_t _version = 1;
 };
 
-void BinaryReportFile::mapForRead( const std::string& path )
+class BinaryReportMap
 {
-    _map.remap( path );
-    const size_t totalSize = _map.getSize();
-
-    if( totalSize < sizeof( Header ) ||
-        ( totalSize % sizeof( uint32_t ) ) != 0 )
+public:
+    // Read-only mapping
+    BinaryReportMap( const std::string& path )
+        : _map( path )
     {
-        LBTHROW( std::runtime_error( "Incompatible binary report: " + path ) );
+        const size_t totalSize = _map.getSize();
+        if( totalSize < sizeof( Header ) ||
+            ( totalSize % sizeof( uint32_t ) ) != 0 )
+        {
+            LBTHROW( std::runtime_error( "Incompatible binary report: " + path ));
+        }
+
+        if( !_map.getAddress< Header >()->isValid( ))
+        {
+            LBTHROW( std::runtime_error( "Invalid binary spike report header: " + path ) );
+        }
     }
 
-    const Header* header = _map.getAddress< Header >();
-    if ( !header->isValid( ))
+    // read-write mapping
+    BinaryReportMap( const std::string& path, size_t nSpikes )
+        : _map( path, sizeof( Header ) + sizeof( Spike ) * nSpikes )
     {
-        LBTHROW( std::runtime_error( "Invalid binary spike report header: " + path ) );
+        *(_map.getAddress< Header >( )) = Header();
     }
 
-    _mappedForRead = true;
-}
-
-void BinaryReportFile::mapForReadWrite( const std::string& path,
-                                        const size_t spikeCount )
-{
-    const size_t newSize = sizeof( Header ) + sizeof( Spike ) * spikeCount;
-    if( _map.getAddress && !_mappedForRead )
-        _map.resize( newSize )
-    else
-        _map.recreate( path, newSize );
+    void resize( const size_t nSpikes )
+    {
+        _map.resize( sizeof( Header ) + sizeof( Spike ) * nSpikes );
+    }
 
-    *(_map.getAddress< Header >( )) = Header();
-    _mappedForRead = false;
-}
+    size_t getNSpikes() const
+    {
+        return ( _map.getSize() - sizeof( Header ) ) / sizeof( Spike );
+    }
 
-size_t BinaryReportFile::getSpikeCount() const
-{
-    return ( _map.getSize() - sizeof( Header ) ) / sizeof( Spike );
-}
+    const Spike* readSpikes() const
+    {
+        return reinterpret_cast< const Spike* >( _map.getAddress< uint8_t >() +
+                                                 sizeof( Header ));
+    }
 
-Spike* BinaryReportFile::getSpikes() const
-{
-    return reinterpret_cast< Spike* >( _map.getAddress< uint8_t >() +
-                                       sizeof( Header ));
-}
+    Spike* writeSpikes()
+    {
+        return reinterpret_cast< Spike* >( _map.getAddress< uint8_t >() +
+                                           sizeof( Header ));
+    }
```

Comment:
Rename to mutableSpikes() and constSpikes() or use getSpikes for both with const and mutable overloads.